overlord/ifacestate/handlers.go

```diff
@@ -303,3 +306,24 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	setConns(st, conns)
 	return nil
 }
+
+func (m *InterfaceManager) doDiscardNamespace(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapSetup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapSetup.Name()
+
+	// Run "snap-discard-ns $SNAP_NAME" and ignore the error code.  This
+	// command is not meant to always succeed because the namespace may not
+	// have been created yet in practice (e.g. the snap was removed before it
```

Comment:
I think  you are right. I've patched snap-discard-ns since writing this code to be resilient to this error. I'll make sure we actually check for errors in this task.